app/public/cantusdata/management/commands/index_manuscript_mei.py

```diff
@@ -0,0 +1,99 @@
+from typing import Any, Dict
+from os import path, listdir
+
+from django.core.management.base import BaseCommand, CommandParser
+from django.conf import settings
+from cantusdata.helpers.mei_processing.mei_tokenizer import MEITokenizer
+from cantusdata.models.folio import Folio
+
+from solr.core import SolrConnection  # type: ignore
+
+MEI4_DIR = path.join("/code", "production-mei-files")
+
+
+class Command(BaseCommand):
+    help = (
+        "This command indexes the contents of MEI files in Solr, using"
+        "the MEITokenizer class to extract n-grams from the MEI files."
+        "Files must be named in the format [some string]_[folio number].mei."
+    )
+
+    def add_arguments(self, parser: CommandParser) -> None:
+        parser.add_argument(
+            "manuscript_id",
+            type=int,
+            nargs=1,
+            help=(
+                "The ID of the manuscript whose MEI data should be indexed."
+                "Must have MEI files in a subdirectory of the --mei-dir argument"
+                "named with this ID."
+            ),
+        )
+        parser.add_argument(
+            "--mei-dir",
+            type=str,
+            default=MEI4_DIR,
+            help=(
+                "The directory containing the MEI files to be indexed."
+                "Defaults to '/code/production-mei-files'."
+            ),
+        )
+        parser.add_argument(
+            "--min-ngram",
+            type=int,
+            default=1,
+            help="The minimum n-gram length to index from the MEI files.",
+        )
+        parser.add_argument(
+            "--max-ngram",
+            type=int,
+            default=5,
+            help="The maximum n-gram length to index from the MEI files.",
+        )
+        parser.add_argument(
+            "--flush-index",
+            action="store_true",
+            help=(
+                "If this flag is set, the command will delete all existing OMR"
+                "documents for the specified manuscript before indexing the new data."
+            ),
+        )
+
+    def handle(self, *args: Any, **options: Any) -> None:
+        solr_conn = SolrConnection(settings.SOLR_SERVER)
+        manuscript_id = options["manuscript_id"][0]
+        if options.get("flush_index"):
+            self.flush_manuscript_ngrams_from_index(solr_conn, manuscript_id)
+        else:
+            folio_map: Dict[str, str] = dict(
+                Folio.objects.filter(manuscript_id=manuscript_id).values_list(
+                    "number", "image_uri"
+                )
+            )
+            manuscript_mei_path = path.join(options["mei_dir"], str(manuscript_id))
+            manuscript_mei_files = [
+                f for f in listdir(manuscript_mei_path) if f.endswith(".mei")
+            ]
```

Comment:
There should maybe be an exception here that checks if the path exists in case `--mei-dir` is formatted incorrectly.